### PR TITLE
Instaredis support for go-redis v9

### PR DESCRIPTION
**Pull request by sanojsubran**
This PR introduces the major version upgrade for instaredis to v2. Support has been added for the changes in go-redis v9.

### CODE REVIEW

File: instrumentation/instaredis/v2/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2022 IBM Corp.
```
Comment (willianpc) [inline instrumentation/instaredis/v2/LICENSE.md:3]: Change year to 2023 please.

File: instrumentation/instaredis/v2/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+instaredis.WrapClient(rdb, sensor)
+```
+
+
+See the [`instaredis` package documentation][godoc] for detailed examples.
+
+
+[godoc]: https://pkg.go.dev/github.com/instana/go-sensor/instrumentation/instaredis/v2
+[instaredis.WrapClient]: https://pkg.go.dev/github.com/instana/go-sensor/instrumentation/instaredis#WrapClient
```
Comment (willianpc) [inline instrumentation/instaredis/v2/README.md:39]: I think this should be `instaredis/v2#WrapClient`, no?

File: instrumentation/instaredis/v2/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+[instaredis.WrapClusterClient]: https://pkg.go.dev/github.com/instana/go-sensor/instrumentation/instaredis#WrapClusterClient
```
Comment (willianpc) [inline instrumentation/instaredis/v2/README.md:40]: Same thing here. to add `/v2`

File: instrumentation/instaredis/v2/example_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+// (c) Copyright IBM Corp. 2022
```
Comment (willianpc) [inline instrumentation/instaredis/v2/example_test.go:1]: Please, replace 2022 by 2023.

File: instrumentation/instaredis/v2/instaredis_test.go [hunk 5]
```diff
@@ -0,0 +1,510 @@
+// (c) Copyright IBM Corp. 2022
```
Comment (willianpc) [inline instrumentation/instaredis/v2/instaredis_test.go:1]: Same thing with the year :)

File: instrumentation/instaredis/v2/version.go [hunk 6]
```diff
@@ -0,0 +1,6 @@
+// (c) Copyright IBM Corp. 2022
```
Comment (willianpc) [inline instrumentation/instaredis/v2/version.go:1]: And again :D

**Review by willianpc — APPROVED** [review]
Looks great! Good job, Sanoj!

Most of my comments are about the copyright year, but you may want to double check my comment in the README.md file :)

Cheers!

File: instrumentation/instaredis/v2/LICENSE.md (hunk 1 above)
Comment (sanojsubran) [reply instrumentation/instaredis/v2/LICENSE.md]: Updated it.

File: instrumentation/instaredis/v2/README.md (hunk 2 above)
Comment (sanojsubran) [reply instrumentation/instaredis/v2/README.md]: Updated it.

File: instrumentation/instaredis/v2/example_test.go (hunk 4 above)
Comment (sanojsubran) [reply instrumentation/instaredis/v2/example_test.go]: Updated it.

File: instrumentation/instaredis/v2/instaredis_test.go (hunk 5 above)
Comment (sanojsubran) [reply instrumentation/instaredis/v2/instaredis_test.go]: Updated it.

File: instrumentation/instaredis/v2/version.go (hunk 6 above)
Comment (sanojsubran) [reply instrumentation/instaredis/v2/version.go]: Updated it.

**Pull request merged**